app/models/project.rb

```diff
@@ -56,9 +56,8 @@ def available_extensions
   end
 
   def extensions
-    attributes["extensions"] ||= []
-    attributes["extensions"].reject!(&:blank?)
-    attributes["extensions"]
+    attributes["extensions"].reject!(&:blank?) if attributes["extensions"]
+    attributes["extensions"] || []
```

Comment:
Oh, I see what you're doing. Instead of updating the extensions to the empty list, you're just returning an empty list. Unfortunately, it's now indeterminate whether modifying that return value will modify the extensions of the project — when your extensions evaluate to `true`, the returned list can be modified in order to modify the attributes; if the extensions evaluate to `false`, they can't.

Here's the test you need:
```ruby
project = Project.new
project.extensions.push("foo")
assert project.extensions.include? "foo" # Should be true, is currently false
project.attributes["extensions"] = ["bar"]
project.extensions.push("foo")
assert project.extensions.include? "foo" # Should be true, I expect it to be true presently
``

This code is the problem:
```
project.extensions.push("foo")
project.extensions.include? "foo"
```

Whether or not the second line returns `true` depends on whether `project.extensions` is empty. That's a bad API.